inc/AI.hpp

```diff
@@ -0,0 +1,108 @@
+/*
+** EPITECH PROJECT, 2024
+** *
+** File description:
+** AI
+*/
+
+#ifndef AI_HPP
+    #define AI_HPP
+
+    #include <iostream>
+    #include "Stone.hpp"
+    #include "Board.hpp"
+
+namespace Gomoku {
+    /**
+     * @brief Class for the AI
+     *
+     * This class contains the AI of the game
+     */
+    class AI {
+        public:
+            /**
+             * @brief Construct a new AI object
+             *
+             */
+            AI(Board _board) : _board(_board)
+            {
+                _color = Color::NONE;
+            }
+
+            /**
+             * @brief Destroy the AI object
+             *
+             */
+            ~AI();
+
+            /**
+             * @brief Set the color of the AI
+             *
+             * @param Color color
+             */
+            void setColor(Color color)
+            {
+                _color = color;
+            }
+
+            /**
+             * @brief Get the color of the AI
+             *
+             * @return Color
+             */
+            Color getColor(void) const
```

Comment:
For this function, i meant something more like : 

If we give it an int equal to 1, then we return our color, else we return the enemy color.

This is important for the function Board most importantly since, as you can read from it : 

[field] is either number 1 (own stone) or number 2 (opponent's stone) or number 3 (only if continuous game is enabled, stone is part of winning line or is forbidden according to renju rules). 

So we need to handle the case where 3 is given as well